p2p/protocol/identify/id_push.go

```diff
@@ -13,5 +14,7 @@ const IDPush = "/ipfs/id/push/1.0.0"
 
 // pushHandler handles incoming identify push streams. The behaviour is identical to the ordinary identify protocol.
 func (ids *IDService) pushHandler(s network.Stream) {
+	s.SetReadDeadline(time.Now().Add(StreamReadTimeout))
+
 	ids.handleIdentifyResponse(s)
```

Comment:
Don't we already set a read deadline in this function?

Reply:
Have fixed this.